### ISSUE TITLE
[Bug]: ansible_netbox.netbox.netbox_virtual_machine crashes with custom_dict.items issue.

### ISSUE THREAD

**Issue by damo2929** [netbox-community/ansible_modules#924]
### Ansible NetBox Collection version

v3.10.0

### Ansible version

ansible [core 2.13.3]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /home/ansible/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.14 (main, Nov  7 2022, 00:00:00) [GCC 11.3.1 20220421 (Red Hat 11.3.1-2)]
  jinja version = 3.1.2
  libyaml = True

ansible-galaxy collection list

# /home/ansible/.ansible/collections/ansible_collections
Collection           Version
-------------------- -------
ansible.netcommon    4.1.0
ansible.posix        1.4.0
ansible.utils        2.8.0
ansible.windows      1.12.0
azure.azcollection   1.14.0
community.crypto     2.9.0
community.general    6.0.1
community.mysql      3.5.1
community.postgresql 2.3.1
community.proxysql   1.4.0
community.zabbix     1.9.0
netbox.netbox        3.10.0

### NetBox version

v3.3.9

### Python version

3.9

### Steps to Reproduce

trying work on any action for the virtual machine module

have customer fields in place as below example machine from the API
```
{
  "count": 1,
  "next": null,
  "previous": null,
  "results": [
    {
      "id": 64,
      "url": "https://10.199.98.220/api/virtualization/virtual-machines/64/",
      "display": "tester",
      "name": "tester",
      "status": {
        "value": "active",
        "label": "Active"
      },
      "site": {
        "id": 1,
        "url": "https://10.199.98.220/api/dcim/sites/1/",
        "display": "Wallingford",
        "name": "Wallingford",
        "slug": "wallingford"
      },
      "cluster": {
        "id": 1,
        "url": "https://10.199.98.220/api/virtualization/clusters/1/",
        "display": "wl_scicom",
        "name": "wl_scicom"
      },
      "device": null,
      "role": null,
      "tenant": {
        "id": 1,
        "url": "https://10.199.98.220/api/tenancy/tenants/1/",
        "display": "Science Computing",
        "name": "Science Computing",
        "slug": "science_computing"
      },
      "platform": null,
      "primary_ip": null,
      "primary_ip4": null,
      "primary_ip6": null,
      "vcpus": 5,
      "memory": 8192,
      "disk": 50,
      "comments": "",
      "local_context_data": null,
      "tags": [],
      "custom_fields": {
        "proxmox_id": null,
        "zabbixid": null,
        "pool": "prodlinux_ec",
        "proxmox_type": "qemu",
        "OS": "Rocky 8",
        "adjoined": null,
        "defaultvlan": {
          "id": 18,
          "url": "https://10.199.98.220/api/ipam/vlans/18/",
          "display": "Linux Dev Network 1 (422)",
          "vid": 422,
          "name": "Linux Dev Network 1"
        }
      },
      "config_context": {},
      "created": "2023-01-10T11:09:45.890449Z",
      "last_updated": "2023-01-10T11:09:45.890475Z"
    }
  ]
}
```

playbook actions tested 

```
  - name: remove machine from netbox.
      local_action:
         module: netbox.netbox.netbox_virtual_machine
         netbox_url: "{{ netbox_endpoint }}"
         netbox_token: "{{ netbox_token }}"
         validate_certs: "{{ netbox_valid_cert }}"
         data:
           name: "{{ inventory_hostname }}"
         state: absent
```

```
     - name: update state and add primary ipv4 address
       local_action:
          module: netbox.netbox.netbox_virtual_machine
          netbox_url: "{{ netbox_endpoint }}"
          netbox_token: "{{ netbox_token }}"
          validate_certs: "{{ netbox_valid_cert }}"
          data:
              name: "{{ inventory_hostname }}"
              primary_ip4: "{{ ipallocation.ip_address.address }}"
             status: active
          state: present
```

### Expected Behavior

successful update.

### Observed Behavior

`
fatal: [newtwo -> localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "/usr/lib/python3.9/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.199.98.220'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/lib/python3.9/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.199.98.220'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/lib/python3.9/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.199.98.220'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  warnings.warn(\n/usr/lib/python3.9/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.199.98.220'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  warnings.warn(\nTraceback (most recent call last):\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1673348735.091881-22767-6253260771134/AnsiballZ_netbox_virtual_machine.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1673348735.091881-22767-6253260771134/AnsiballZ_netbox_virtual_machine.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/ansible/.ansible/tmp/ansible-tmp-1673348735.091881-22767-6253260771134/AnsiballZ_netbox_virtual_machine.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.netbox.netbox.plugins.modules.netbox_virtual_machine', init_globals=dict(_module_fqn='ansible_collections.netbox.netbox.plugins.modules.netbox_virtual_machine', _modlib_path=modlib_path),\n  File \"/usr/lib64/python3.9/runpy.py\", line 225, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib64/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_netbox.netbox.netbox_virtual_machine_payload_rb447ogg/ansible_netbox.netbox.netbox_virtual_machine_payload.zip/ansible_collections/netbox/netbox/plugins/modules/netbox_virtual_machine.py\", line 243, in <module>\n  File \"/tmp/ansible_netbox.netbox.netbox_virtual_machine_payload_rb447ogg/ansible_netbox.netbox.netbox_virtual_machine_payload.zip/ansible_collections/netbox/netbox/plugins/modules/netbox_virtual_machine.py\", line 239, in main\n  File \"/tmp/ansible_netbox.netbox.netbox_virtual_machine_payload_rb447ogg/ansible_netbox.netbox.netbox_virtual_machine_payload.zip/ansible_collections/netbox/netbox/plugins/module_utils/netbox_virtualization.py\", line 74, in run\n  File \"/usr/lib/python3.9/site-packages/pynetbox/core/response.py\", line 457, in serialize\n    ret[i] = flatten_custom(current_val)\n  File \"/usr/lib/python3.9/site-packages/pynetbox/core/response.py\", line 61, in flatten_custom\n    return {\n  File \"/usr/lib/python3.9/site-packages/pynetbox/core/response.py\", line 62, in <dictcomp>\n    k: v if not isinstance(v, dict) else v[\"value\"] for k, v in custom_dict.items()\nKeyError: 'value'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
`

**Comment by rodvand** [comment]
Should be fixed if you update to pynetbox 7.0.1 and collection 3.11.0.

**Comment by damo2929** [comment]
it has worked thank you.
